providers/implementations/storemgmt/file_store.c

```diff
@@ -437,6 +437,31 @@ static int file_setup_decoders(struct file_ctx_st *ctx)
             goto err;
         }
 
+        /*
+         * Where applicable, set the outermost structure name.
+         * The goal is to avoid the STORE object types that are
+         * potentially password protected but aren't interesting
+         * for this load.
+         */
+        switch (ctx->expected_type) {
+        case OSSL_STORE_INFO_CERT:
+            if (!OSSL_DECODER_CTX_set_input_structure(ctx->_.file.decoderctx,
+                                                      "Certificate")) {
+                ERR_raise(ERR_LIB_PROV, ERR_R_OSSL_DECODER_LIB);
+                goto err;
+            }
+            break;
+        case OSSL_STORE_INFO_CRL:
+            if (!OSSL_DECODER_CTX_set_input_structure(ctx->_.file.decoderctx,
+                                                      "CertificateList")) {
```

Comment:
Should this be CRL instead? Or `X509 CRL` to match with the PEM header for CRLs?

Reply:
I've use the structure names from the ASN.1 modules.  That's why you see things like "SubjectPublicKeyInfo", "PrivateKeyInfo", "EncryptedPrivateKeyInfo", ...  "Certificate" and "CertificateList" follows that line of thought.

Reply:
We did at some point think about using the PEM names as structure names, but never went through with it.  It would probably be quite disruptive at this point...

Reply:
OK then.